x-pack/legacy/plugins/siem/public/components/super_date_picker/selectors.test.ts

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  policySelector,
+  durationSelector,
+  kindSelector,
+  startSelector,
+  endSelector,
+  fromStrSelector,
+  toStrSelector,
+  isLoadingSelector,
+  refetchSelector,
+} from './selectors';
+import { InputsRange, AbsoluteTimeRange, RelativeTimeRange } from '../../store/inputs/model';
+import { cloneDeep } from 'lodash/fp';
+
+describe('selectors', () => {
+  let absoluteTime: AbsoluteTimeRange = {
+    kind: 'absolute',
+    fromStr: undefined,
+    toStr: undefined,
+    from: 0,
+    to: 0,
+  };
+
+  let inputState: InputsRange = {
+    timerange: absoluteTime,
+    policy: {
+      kind: 'manual',
+      duration: 0,
+    },
+    query: [],
+    linkTo: [],
+  };
+
+  const getPolicySelector = policySelector();
+  const getDurationSelector = durationSelector();
+  const getKindSelector = kindSelector();
+  const getStartSelector = startSelector();
+  const getEndSelector = endSelector();
+  const getFromStrSelector = fromStrSelector();
+  const getToStrSelector = toStrSelector();
+  const getIsLoadingSelector = isLoadingSelector();
+  const getRefetchSelector = refetchSelector();
+
+  beforeEach(() => {
+    absoluteTime = {
+      kind: 'absolute',
+      fromStr: undefined,
+      toStr: undefined,
+      from: 0,
+      to: 0,
+    };
+
+    inputState = {
+      timerange: absoluteTime,
+      policy: {
+        kind: 'manual',
+        duration: 0,
+      },
+      query: [],
+      linkTo: [],
+    };
+  });
+
+  describe('#policySelector', () => {
+    test('returns the same reference given the same identical input twice', () => {
+      const result1 = getPolicySelector(inputState);
+      const result2 = getPolicySelector(inputState);
+      expect(result1).toBe(result2);
+    });
+
+    test('returns the same reference given different input twice but with different deep copies', () => {
+      const clone = cloneDeep(inputState);
+      const result1 = getPolicySelector(inputState);
+      const result2 = getPolicySelector(clone);
+      expect(result1).toBe(result2);
+    });
+
+    test('returns a different reference given different policy kind', () => {
+      const result1 = getPolicySelector(inputState);
+      const change: InputsRange = {
+        ...inputState,
+        policy: { ...inputState.policy, ...{ kind: 'interval' } },
```

Comment:
optional: i think you may omit the extra spread and braces, i.e.  `...{ kind: 'interval' }`, as the following should be equivalent syntax:

```
policy: { ...inputState.policy, kind: 'interval' },
```

(this comment, if optionally implemented, is applicable to similar tests in this pr)

Reply:
Nice, I implemented this everywhere.